src/CoreFx.Private.TestUtilities/src/System/Diagnostics/RemoteExecutorTestBase.uap.cs

```diff
@@ -62,15 +67,60 @@ private static RemoteInvokeHandle RemoteInvoke(MethodInfo method, string[] args,
                     i++;
                 }
 
-                AppServiceResponse response = remoteExecutionService.SendMessageAsync(message).GetAwaiter().GetResult();
+                asyncOperation = remoteExecutionService.SendMessageAsync(message);
+            }
+            finally
+            {
+                if (asyncOperation == null)
+                {
+                    remoteExecutionService.Dispose();
+                }
+            }
+
+            // Wait synchronously in a background thread so as not to block the test
+            Action waitForRemoteWaitThread;
+            Thread remoteWaitThread = ThreadTestHelpers.CreateGuardedThread(out waitForRemoteWaitThread, () =>
+            {
+                using (remoteExecutionService)
+                {
+                    AppServiceResponse response = asyncOperation.GetAwaiter().GetResult();
```

Comment:
Decided to move all of the original code into the background thread. Can't get errors early after the first blocking operation, so any errors after that point will only be found at the RemoteInvokeHandle's Dispose call.